File: codegen/results/v002/qwen3.6_35b-a3b-coding-nvfp4_nothink/20260418_194110/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const filePath = path.join(process.cwd(), 'input', 'text.txt');
const content = fs.readFileSync(filePath, 'utf-8');

// Convert to lowercase
const lowerContent = content.toLowerCase();

// Split into words: split by non-letter characters
// Replace all non-letter characters with spaces, then split by whitespace
const cleaned = lowerContent.replace(/[^a-z]/g, ' ');
const words = cleaned.split(/\s+/).filter(word => word.length > 0);

// Count frequencies
const freq: Record<string, number> = {};
for (const word of words) {
  freq[word] = (freq[word] || 0) + 1;
}

// Get unique words
const uniqueWords = Object.keys(freq);

// Sort: by count descending, then alphabetically ascending
uniqueWords.sort((a, b) => {
  const countDiff = freq[b] - freq[a];
  if (countDiff !== 0) return countDiff;
  return a.localeCompare(b);
});

// Output
for (const word of uniqueWords) {
  console.log(`${word}: ${freq[word]}`);
}

File: codegen/results/v002/qwen3.6_35b-a3b-coding-nvfp4_nothink/20260418_194110/python/002_word_frequency/solution.py
import re
from collections import Counter

def main():
    with open('input/text.txt', 'r') as f:
        text = f.read()
    
    # Convert to lowercase
    text = text.lower()
    
    # Replace all non-letter characters with spaces
    # This effectively strips punctuation and splits on any non-letter
    cleaned_text = re.sub(r'[^a-z]', ' ', text)
    
    # Split into words
    words = cleaned_text.split()
    
    # Filter out empty strings (if any)
    words = [w for w in words if w]
    
    # Count frequencies
    counter = Counter(words)
    
    # Sort by count descending, then alphabetically ascending for ties
    sorted_words = sorted(counter.items(), key=lambda x: (-x[1], x[0]))
    
    # Output
    for word, count in sorted_words:
        print(f"{word}: {count}")

if __name__ == '__main__':
    main()